plugins/flytekit-kf-mpi/flytekitplugins/kfmpi/task.py

```diff
@@ -110,34 +140,110 @@ class MPIFunctionTask(PythonFunctionTask[MPIJob]):
     ]
 
     def __init__(self, task_config: MPIJob, task_function: Callable, **kwargs):
+        if task_config.num_workers and task_config.worker.replicas:
+            raise ValueError(
+                "Cannot specify both `num_workers` and `worker.replicas`. Please use `worker.replicas` as `num_workers` is depreacated."
+            )
+        if task_config.num_workers is None and task_config.worker.replicas is None:
+            raise ValueError(
+                "Must specify either `num_workers` or `worker.replicas`. Please use `worker.replicas` as `num_workers` is depreacated."
+            )
+        if task_config.num_launcher_replicas and task_config.launcher.replicas:
+            raise ValueError(
+                "Cannot specify both `num_workers` and `launcher.replicas`. Please use `launcher.replicas` as `num_launcher_replicas` is depreacated."
+            )
+        if task_config.num_launcher_replicas is None and task_config.launcher.replicas is None:
+            raise ValueError(
+                "Must specify either `num_workers` or `launcher.replicas`. Please use `launcher.replicas` as `num_launcher_replicas` is depreacated."
+            )
         super().__init__(
             task_config=task_config,
             task_function=task_function,
             task_type=self._MPI_JOB_TASK_TYPE,
             **kwargs,
         )
 
+    def _convert_replica_spec(
+        self, replica_config: Union[Launcher, Worker]
+    ) -> mpi_task.DistributedMPITrainingReplicaSpec:
+        resources = convert_resources_to_resource_model(requests=replica_config.requests, limits=replica_config.limits)
+        return mpi_task.DistributedMPITrainingReplicaSpec(
+            command=replica_config.command,
+            replicas=replica_config.replicas,
+            image=replica_config.image,
+            resources=resources.to_flyte_idl() if resources else None,
+            restart_policy=replica_config.restart_policy.value if replica_config.restart_policy else None,
+        )
+
+    def _convert_run_policy(self, run_policy: RunPolicy) -> kubeflow_common.RunPolicy:
+        return kubeflow_common.RunPolicy(
+            clean_pod_policy=run_policy.clean_pod_policy.value if run_policy.clean_pod_policy else None,
+            ttl_seconds_after_finished=run_policy.ttl_seconds_after_finished,
+            active_deadline_seconds=run_policy.active_deadline_seconds,
+            backoff_limit=run_policy.active_deadline_seconds,
+        )
+
+    def _get_base_command(self, settings: SerializationSettings) -> List[str]:
```

Comment:
just curious, why do we need this method?

Reply:
Good question. So HorovodFunctionTask extends MPIFunctionTask extends PythonFunctionTask, and get_command get overriden by MPIFunctionTask.

In HorovodFunctionTask, I would like to use the get_command() from PythonFunctionTask, thus I have to create yet another function in MPIFunctionTask to do so.